clang-tools-extra/test/clang-tidy/checkers/modernize/use-nullptr.cpp

```diff
@@ -84,6 +84,14 @@ void test_macro_expansion4() {
 #undef MY_NULL
 }
 
+template <typename T> struct pear {
+  T x;
+};
+void test_templated() {
+  pear<int*> p = { NULL };
```

Comment:
Doesn't this need some `CHECK` or `CHECK-NOT`?

Reply:
It definitely does, but I was hoping that the absence would cause some kind of test failure! I.e. "first make sure you see the test fail"... Let me add a bogus expectation to confirm.

Reply:
Done now. A key insight was that the test defines `NULL` as `0`, so in order for the new check to kick in, we're using `__null` as the to-be-diagnosed spelling.